File: src/main/java/org/smarthata/service/CronService.java
package org.smarthata.service;

import org.smarthata.service.message.SmarthataMessage;
import org.smarthata.service.message.SmarthataMessageBroker;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.scheduling.annotation.Scheduled;
import org.springframework.stereotype.Service;

import static org.smarthata.service.message.EndpointType.*;

@Service
public class CronService {

    private final SmarthataMessageBroker messageBroker;
    private final WeatherService weatherService;

    @Autowired
    public CronService(SmarthataMessageBroker messageBroker, WeatherService weatherService) {
        this.messageBroker = messageBroker;
        this.weatherService = weatherService;
    }

    @Scheduled(cron = "0 */5 * * * *")
    public void calcAverageDailyStreetTemp() {
        double averageDailyStreetTemperature = weatherService.calcAverageDailyStreetTemperature();

        SmarthataMessage message = new SmarthataMessage("/street/temp-average", Double.toString(averageDailyStreetTemperature), SYSTEM, MQTT);
        messageBroker.broadcastSmarthataMessageRetained(message);
    }

    @Scheduled(cron = "0 */5 * * * *")
    public void sendDataToNarodmon() {
        weatherService.sendDataToNarodmon();
    }

}

File: src/main/kotlin/org/smarthata/service/CronService.kt
package org.smarthata.service

import org.smarthata.service.message.EndpointType
import org.smarthata.service.message.SmarthataMessage
import org.smarthata.service.message.SmarthataMessageBroker
import org.springframework.scheduling.annotation.Scheduled
import org.springframework.stereotype.Service

@Service
class CronService(
    private val messageBroker: SmarthataMessageBroker,
    private val weatherService: WeatherService,
    private val sensorCleaner: SensorCleaner,
) {
    @Scheduled(cron = "0 */1 * * * *")
    fun calcAverageStreetTemp() {
        messageBroker.broadcast(
            smarthataMessage(
                weatherService.calcAverageDailyStreetTemperature(),
                "/street/temp-average"
            )
        )
        messageBroker.broadcast(
            smarthataMessage(
                weatherService.calcAverageWeeklyStreetTemperature(),
                "/street/temp-weekly-average"
            )
        )
    }

    private fun smarthataMessage(value: Double, topic: String) = SmarthataMessage(
        topic,
        value.toString(),
        EndpointType.SYSTEM,
        EndpointType.MQTT,
        true
    )

    @Scheduled(cron = "0 0 4 * * *")
    fun cleanSensorsData() {
        sensorCleaner.cleanAllDevices()
    }
}
